ext2/ext2.c

```diff
@@ -568,22 +604,33 @@ static int _put_blkno(ext2_t* ext2, uint32_t blkno)
     ext2->groups[grpno].bg_free_blocks_count++;
 
     /* write the group and bitmap */
-    ECHECK(_write_group_with_bitmap(ext2, grpno, &bitmap));
+    ECHECK(_write_group_with_bitmap(ext2, grpno, bitmap));
 
     /* ATTN: minimize super block writes */
     /* update super block. */
     ECHECK(_write_super_block(ext2));
 
 done:
+
+    if (bitmap)
+        free(bitmap);
+
     return ret;
 }
 
 static int _get_blkno(ext2_t* ext2, uint32_t* blkno)
 {
     int ret = 0;
-    ext2_block_t bitmap;
+    struct vars
+    {
```

Comment:
Define `struct vars` at the file scope so we don't have to define it over and over again.

Reply:
The definition varies from function to function.